app/com/arpnetworking/metrics/portal/scheduling/impl/DatabaseExecutionHelper.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright 2020 Dropbox, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.arpnetworking.metrics.portal.scheduling.impl;
+
+import com.arpnetworking.metrics.portal.scheduling.JobExecutionRepository;
+import com.arpnetworking.steno.Logger;
+import com.google.common.base.Throwables;
+import io.ebean.EbeanServer;
+import io.ebean.Transaction;
+import models.ebean.BaseExecution;
+import models.internal.Organization;
+import models.internal.scheduling.JobExecution;
+
+import java.time.Instant;
+import java.util.UUID;
+import java.util.function.Consumer;
+import javax.annotation.Nullable;
+import javax.persistence.PersistenceException;
+
+/**
+ * Helper class for implementing a SQL-backed {@link JobExecutionRepository}, providing facilities for updating the state
+ * of an execution as well as mapping it back to an internal representation.
+ * <p>
+ * Classes using this should extend {@link BaseExecution} and delegate their {@code jobXXX} calls to this helper.
+ *
+ * @param <T> The type of result produced by each job.
+ * @param <E> The type of the repository's bean.
+ *
+ * @author Christian Briones (cbriones at dropbox dot com)
+ */
+public final class DatabaseExecutionHelper<T, E extends BaseExecution<T>> {
+    private final EbeanServer _ebeanServer;
+    private final ExecutionAdapter<T, E> _adapter;
+    private final Logger _logger;
+
+    /**
+     * Public constructor.
+     *
+     * @param logger The logger for the repository.
+     * @param ebeanServer An ebean server.
+     * @param adapter The execution adapter for the repository.
+     */
+    public DatabaseExecutionHelper(
+            final Logger logger,
```

Comment:
Not sure if it's frowned upon to pass in a logger like this, but it seemed like the only way to share the code in `updateExecutionState` without declaring multiple methods (at which point I don't think it'd be worth it).